clippy_lints/src/lifetimes.rs

```diff
@@ -639,44 +556,49 @@ where
 }
 
 fn report_extra_lifetimes<'tcx>(cx: &LateContext<'tcx>, func: &'tcx FnDecl<'_>, generics: &'tcx Generics<'_>) {
-    let hs = generics
+    let def_ids = generics
         .params
         .iter()
         .filter_map(|par| match par.kind {
             GenericParamKind::Lifetime {
                 kind: LifetimeParamKind::Explicit,
-            } => Some((par.name.ident().name, par.span)),
+            } => Some(par.def_id),
             _ => None,
         })
         .collect();
-    let mut checker = LifetimeChecker::<hir_nested_filter::None>::new(cx, hs);
+    let mut checker = LifetimeChecker::<hir_nested_filter::None>::new(cx, def_ids);
```

Comment:
We could collect into the `HashMap` `LifetimeChecker` wants directly rather than creating a temporary `Vec`

Reply:
Maybe pass the `Generics` into `LifetimeChecker::new` and have the collect done in there? (Maybe this is what you were suggesting.)

Reply:
I was just thinking of passing in `FxHashMap<LocalDefId, Vec<Usage>>` but yeah that's better since there's two places the collect is being done already